ibm/config.go

```diff
@@ -594,91 +594,368 @@ func (sess clientSession) IAMNamespaceAPI() (*ns.IbmCloudFunctionsNamespaceAPIV1
+			Crn:            core.StringPtr(""),
+			ZoneIdentifier: core.StringPtr(""),
+			Authenticator:  sess.cisDomainSettingsClient.Service.Options.Authenticator,
+		}
+		cisDomainSettingsClient, cisDomainSettingsErr :=
+			cisdomainsettingsv1.NewZonesSettingsV1(cisDomainSettingsOpt)
+		if cisDomainSettingsErr != nil {
+			return nil, fmt.Errorf(
+				"Error occured while configuring CIS Domain Settings client: %s",
+				cisDomainSettingsErr)
+		}
+		return cisDomainSettingsClient, nil
+	}
 	return sess.cisDomainSettingsClient, sess.cisDomainSettingsErr
 }
 
 // CIS Routing
 func (sess clientSession) CisRoutingClientSession() (*cisroutingv1.RoutingV1, error) {
+	if sess.cisRoutingErr == nil {
+		cisRoutingOpt := &cisroutingv1.RoutingV1Options{
+			URL:            sess.cisRoutingClient.Service.Options.URL,
+			Crn:            core.StringPtr(""),
+			ZoneIdentifier: core.StringPtr(""),
+			Authenticator:  sess.cisRoutingClient.Service.Options.Authenticator,
+		}
+		cisRoutingClient, cisRoutingErr :=
+			cisroutingv1.NewRoutingV1(cisRoutingOpt)
+		if cisRoutingErr != nil {
+			return nil, fmt.Errorf(
+				"Error occured while configuring CIS Routing client: %s",
+				cisRoutingErr)
+		}
+		return cisRoutingClient, nil
+	}
 	return sess.cisRoutingClient, sess.cisRoutingErr
 }
 
 // CIS Cache service
 func (sess clientSession) CisCacheClientSession() (*ciscachev1.CachingApiV1, error) {
+	if sess.cisCacheErr == nil {
+		cisCacheOpt := &ciscachev1.CachingApiV1Options{
+			URL:           sess.cisCacheClient.Service.Options.URL,
+			Crn:           core.StringPtr(""),
+			ZoneID:        core.StringPtr(""),
+			Authenticator: sess.cisCacheClient.Service.Options.Authenticator,
+		}
+		cisCacheClient, cisCacheErr :=
+			ciscachev1.NewCachingApiV1(cisCacheOpt)
+		if cisCacheErr != nil {
+			return nil, fmt.Errorf(
+				"Error occured while configuring CIS Caching client: %s",
+				cisCacheErr)
+		}
+		return cisCacheClient, nil
+	}
 	return sess.cisCacheClient, sess.cisCacheErr
 }
 
 // CIS Zone Settings
 func (sess clientSession) CisCustomPageClientSession() (*ciscustompagev1.CustomPagesV1, error) {
+	if sess.cisCustomPageErr == nil {
+		cisCustomPageOpt := &ciscustompagev1.CustomPagesV1Options{
+			URL:            sess.cisCustomPageClient.Service.Options.URL,
+			Crn:            core.StringPtr(""),
+			ZoneIdentifier: core.StringPtr(""),
+			Authenticator:  sess.cisCustomPageClient.Service.Options.Authenticator,
+		}
+
+		cisCustomPageClient, cisCustomPageErr :=
+			ciscustompagev1.NewCustomPagesV1(cisCustomPageOpt)
+		if cisCustomPageErr != nil {
+			return nil, fmt.Errorf(
+				"Error occured while configuring CIS Custom Pages client: %s",
+				cisCustomPageErr)
+		}
+		return cisCustomPageClient, nil
+	}
 	return sess.cisCustomPageClient, sess.cisCustomPageErr
 }
 
 // CIS Firewall access rule
 func (sess clientSession) CisAccessRuleClientSession() (*cisaccessrulev1.ZoneFirewallAccessRulesV1, error) {
+	if sess.cisAccessRuleErr == nil {
```

Comment:
Same here:

```go
if sess.cisAccessRuleErr != nil {
	return sess.cisAccessRuleClient, sess.cisAccessRuleErr
}

cisAccessRuleOpt := &cisaccessrulev1.ZoneFirewallAccessRulesV1Options{
	URL:            sess.cisAccessRuleClient.Service.Options.URL,
	Crn:            core.StringPtr(""),
	ZoneIdentifier: core.StringPtr(""),
	Authenticator:  sess.cisAccessRuleClient.Service.Options.Authenticator,
}
cisAccessRuleClient, cisAccessRuleErr :=
	cisaccessrulev1.NewZoneFirewallAccessRulesV1(cisAccessRuleOpt)
if cisAccessRuleErr != nil {
	return nil, fmt.Errorf(
		"Error occured while configuring CIS Firewall Access Rule client: %s",
		cisAccessRuleErr)
}

return cisAccessRuleClient, nil

```

Reply:
I have upgraded sdk and regenerated cis sdk. Modified the code as per your suggession.